api/tokens.go

```diff
@@ -0,0 +1,124 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/ONSdigital/dp-identity-api/apierrors"
+	"github.com/ONSdigital/dp-identity-api/emailvalidation"
+
+	"github.com/ONSdigital/log.go/log"
+)
+
+type AuthParams struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+var invalidPasswordError = errors.New("Invalid password")
+var invalidPasswordMessage = "Unable to validate the password in the request"
+var invalidEmailError = errors.New("Invalid email")
+var invalidErrorMessage = "Unable to validate the email in the request"
+
+func TokensHandler() http.HandlerFunc {
+
+	return func(w http.ResponseWriter, req *http.Request) {
+
+		ctx := req.Context()
+
+		field := ""
+		param := ""
+
+		defer req.Body.Close()
+
+		body, err := ioutil.ReadAll(req.Body)
+		if err != nil {
+			errorMessage := "api endpoint POST login returned an error reading the request body"
+			handleUnexpectedError(ctx, w, err, errorMessage, field, param)
+			return
+		}
+
+		var authParams AuthParams
+		err = json.Unmarshal(body, &authParams)
+		if err != nil {
+			errorMessage := "api endpoint POST login returned an error unmarshalling the body"
+			handleUnexpectedError(ctx, w, err, errorMessage, field, param)
+			return
+		}
+
+		validPasswordRequest := passwordValidation(authParams)
+		validEmailRequest := emailValidation(authParams)
+		if err != nil {
+			errorMessage := "api endpoint POST login returned an error validating the email"
+			handleUnexpectedError(ctx, w, err, errorMessage, field, param)
+			return
+		}
+
+		invalidPasswordErrorBody := apierrors.IndividualErrorBuilder(invalidPasswordError, invalidPasswordMessage, field, param)
+		invalidEmailErrorBody := apierrors.IndividualErrorBuilder(invalidEmailError, invalidErrorMessage, field, param)
+
+		var errorList []apierrors.IndividualError
+
+		if !validPasswordRequest {
+			errorList = append(errorList, invalidPasswordErrorBody)
+		}
+
+		if !validEmailRequest {
+			errorList = append(errorList, invalidEmailErrorBody)
+		}
+
+		errorResponseBody := apierrors.ErrorResponseBodyBuilder(errorList)
+		writeErrorResponse(ctx, w, 400, errorResponseBody)
```

Comment:
Rather than inlining the `400` response, it's better to use `http.StatusBadRequest` (`net/http` package)

Reply:
This has been updated now, and also a 500 further down